russh-keys/src/format/pkcs8.rs

```diff
@@ -210,6 +269,32 @@ fn write_key_v0_rsa(writer: &mut yasna::DERWriterSeq, key: &Rsa<Private>) {
     writer.next().write_bytes(&bytes);
 }
 
+#[cfg(feature = "openssl")]
+fn read_key(reader: &mut BERReaderSeq) -> Result<Rsa<Private>, Error>{
+    use openssl::bn::BigNum;
+    Ok(Rsa::from_private_components(
+        BigNum::from_slice(&reader.next().read_biguint()?.to_bytes_be())?,
+        BigNum::from_slice(&reader.next().read_biguint()?.to_bytes_be())?,
+        BigNum::from_slice(&reader.next().read_biguint()?.to_bytes_be())?,
+        BigNum::from_slice(&reader.next().read_biguint()?.to_bytes_be())?,
+        BigNum::from_slice(&reader.next().read_biguint()?.to_bytes_be())?,
+        BigNum::from_slice(&reader.next().read_biguint()?.to_bytes_be())?,
+        BigNum::from_slice(&reader.next().read_biguint()?.to_bytes_be())?,
+        BigNum::from_slice(&reader.next().read_biguint()?.to_bytes_be())?,
+    )?)
+}
+
+#[cfg(not(feature = "openssl"))]
+fn read_key(reader: &mut BERReaderSeq) -> Result<RsaPrivateKey, Error> {
+    let n = BigUint::from_bytes_be(&reader.next().read_biguint()?.to_bytes_be());
+    let e = BigUint::from_bytes_be(&reader.next().read_biguint()?.to_bytes_be());
+    let d = BigUint::from_bytes_be(&reader.next().read_biguint()?.to_bytes_be());
+    let _ = &reader.next().read_biguint()?.to_bytes_be();
+    let p = BigUint::from_bytes_be(&reader.next().read_biguint()?.to_bytes_be());
+    let q = BigUint::from_bytes_be(&reader.next().read_biguint()?.to_bytes_be());
+    Ok(RsaPrivateKey::from_components(n, e, d, vec![p, q])?)
```

Comment:
This reads 6 `BigUint`s, which should be 8.
And order is: `n, e, d, p, q, dp, dq, iqmp`

Reply:
oops, thanks!

Reply:
Do we need `dp, dq, iqmp`  for `RsaPrivateKey`? Or will they be pre-computed anyway?

Reply:
My experience with `yasna` is that it won't be happy without consuming everything. I think you'll get the `Extra` error.